arrows-stdlib-cats/src/main/scala/arrows/stdlib/cats.scala

```diff
@@ -0,0 +1,107 @@
+package arrows.stdlib
+
+import cats.{ Applicative, StackSafeMonad }
+import cats.effect.{ Async, ExitCase }
+import cats.arrow.ArrowChoice
+import cats.kernel.{ Monoid, Semigroup }
+import cats.mtl.ApplicativeAsk
+
+import scala.concurrent.Promise
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success }
+
+object Cats extends ArrowInstances
+
+sealed abstract class ArrowInstances extends ArrowInstances1 {
+
+  implicit def catsMonadForArrow[E]: Async[Arrow[E, ?]] = new Async[Arrow[E, ?]] with StackSafeMonad[Arrow[E, ?]] {
+
+    def flatMap[A, B](fa: Arrow[E, A])(f: A => Arrow[E, B]): Arrow[E, B] =
+      Arrow[E].flatMap(e => fa(e).flatMap(a => f(a)(e)))
+
+    def raiseError[A](e: Throwable): Arrow[E, A] = Arrow.failed[A](e)
+
+    def handleErrorWith[A](fa: Arrow[E, A])(f: Throwable => Arrow[E, A]): Arrow[E, A] =
+      Arrow[E].flatMap(e => fa.recoverWith { case t: Throwable => f(t)(e) }(e))
+
+    def pure[A](x: A): Arrow[E, A] = Arrow.successful(x)
+
+    def suspend[A](thunk: => Arrow[E, A]): Arrow[E, A] =
+      Arrow[E].flatMap(e => try { thunk(e) } catch { case NonFatal(t) => Arrow.failed[A](t)(e) })
```

Comment:
you don't need to catch here, `Arrow` will do it

Reply:
Ah, cool thanks! :)